nixos/modules/config/shells-environment.nix

```
@@ -42,8 +42,8 @@ in
         strings.  The latter is concatenated, interspersed with colon
         characters.
       '';
-      type = with types; attrsOf (either str (listOf str));
-      apply = mapAttrs (n: v: if isList v then concatStringsSep ":" v else v);
+      type = with types; attrsOf (oneOf [ str path (listOf str) ]);
+      apply = mapAttrs (n: v: if isList v then concatStringsSep ":" v else "${v}");
```

This logic should be represented by a type instead, so that we can have

```suggestion
      type = with types; attrsOf (envVarSep ":");
```

That will also make it unnecessary to copy this option's internal in the other file.

---

there seems to already be a `types.envVar` that is defined differently and no used anywhere :thinking:

---

@roberth is this good to merge?